server/polar/subscription/service.py

```diff
@@ -568,51 +591,427 @@ async def update_subscription_from_stripe(
         session.add(subscription)
 
         await self.enqueue_benefits_grants(session, subscription)
+        await self._after_subscription_updated(
+            session,
+            subscription,
+            previous_status=previous_status,
+            previous_ends_at=previous_ends_at,
+        )
+        return subscription
+
+    async def update(
+        self,
+        session: AsyncSession,
+        subscription: Subscription,
+        *,
+        updates: SubscriptionUpdate,
+    ) -> Subscription:
+        if isinstance(updates, SubscriptionUpdatePrice):
+            return await self.update_product_price(
+                session,
+                subscription,
+                product_price_id=updates.product_price_id,
+            )
+
+        cancel = updates.cancel_at_period_end is True
+        uncancel = updates.cancel_at_period_end is False
+        if not (updates.revoke or cancel or uncancel):
```

Comment:
I don't understand this case, could use a comment :)

Reply:
Nice catch. Commented, but also changed it to an `any()` to better readability. Just exiting early unless we're asked to revoke, cancel or uncancel.